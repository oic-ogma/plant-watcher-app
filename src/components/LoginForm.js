import React, { Component } from 'react';
import { Label, Error, FormInputComponent } from './common';
import { Actions } from 'react-native-router-flux';
import { View } from 'react-native';
import { Button } from 'react-native-elements';
import { Platform } from 'react-native';

export default class LoginForm extends Component {
  onButtonPress() {
    const { email, password } = this.props;

    this.props.userLogin(email, password);
  }

  renderButtons() {
    return (
      <View>
        <Button style={style.buttonStyle}
          onPress={this.onButtonPress.bind(this)}
          raised
          large
          loading={this.props.loading}
          loadingRight={true}
          title='ログイン'
          backgroundColor='blue'/>

        <Button style={style.buttonStyle}
          onPress={Actions.registration}
          raised
          large
          title='ユーザー登録'
          backgroundColor='#228b22'/>
      </View>
    );
  }

  render() {
    const {
      email,
      password,
      emailChanged,
      passwordChanged,
      error
    } = this.props;

    return (
      <View style={style.margin}>
        <Label placeholder='メールアドレス'/>
        <FormInputComponent
          placeholder='example@hoge.com'
          onChangeText={emailChanged}
          value={email}
        />

        <Label placeholder='パスワード'/>
        <FormInputComponent
          secureTextEntry
          placeholder='パスワード'
          onChangeText={passwordChanged}
          value={password}
        />

        <Error message={error}/>

        {this.renderButtons()}
      </View>
    );
  }
}

const style = {
  buttonStyle: {
    marginBottom: 5,
    ...Platform.select({
      android: {
        marginBottom: 5
      }
    })
  },
  margin: {
    marginTop: 140,
  }
};
