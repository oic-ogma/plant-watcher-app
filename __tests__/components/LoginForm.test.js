import React from 'react';
import LoginForm from '../../src/components/LoginForm';
import renderer from 'react-test-renderer';

describe('ログインフォーム', () => {
  it('正しくrenderできる', () => {
    const rendered = renderer.create(<LoginForm />).toJSON();
    expect(rendered).toBeTruthy();
  });

  it('snapshot-普通', () => {
    const tree = renderer.create(<LoginForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot-メールアドレスとパスワードが入力されている', () => {
    const INITIAL_STATE = {
      email: 'example@gmail.com',
      password: 'PassWord',
      error: '',
      loading: false
    };
    const tree = renderer.create(<LoginForm {...INITIAL_STATE}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot-ロード中', () => {
    const INITIAL_STATE = {
      email: 'example@gmail.com',
      password: 'PassWord',
      error: '',
      loading: true
    };
    const tree = renderer.create(<LoginForm {...INITIAL_STATE}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot-エラーが出ている', () => {
    const INITIAL_STATE = {
      email: 'example@gmail.com',
      password: '',
      error: 'Error',
      loading: true
    };
    const tree = renderer.create(<LoginForm {...INITIAL_STATE}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
