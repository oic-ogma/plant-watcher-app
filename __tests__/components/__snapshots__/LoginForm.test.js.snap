// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ログインフォーム snapshot-エラーが出ている 1`] = `
<View
  style={
    Object {
      "marginTop": 140,
    }
  }
>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            Object {
              "fontSize": 20,
            },
            undefined,
          ],
        ]
      }
    >
      メールアドレス
    </Text>
  </View>
  <View
    collapsable={undefined}
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      onChangeText={undefined}
      placeholder="example@hoge.com"
      secureTextEntry={undefined}
      selectionColor="#86939e"
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          Object {
            "fontSize": 18,
          },
        ]
      }
      value="example@gmail.com"
    />
  </View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            Object {
              "fontSize": 20,
            },
            undefined,
          ],
        ]
      }
    >
      パスワード
    </Text>
  </View>
  <View
    collapsable={undefined}
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      onChangeText={undefined}
      placeholder="パスワード"
      secureTextEntry={true}
      selectionColor="#86939e"
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          Object {
            "fontSize": 18,
          },
        ]
      }
      value=""
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "color": "#FF0000",
        "fontSize": 18,
        "paddingTop": 5,
      }
    }
  >
    Error
  </Text>
  <View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "marginLeft": 15,
            "marginRight": 15,
          },
          Object {
            "shadowColor": "rgba(0,0,0, .4)",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          undefined,
          undefined,
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "marginBottom": 5,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "blue",
              },
              undefined,
              false,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                  undefined,
                  false,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            ログイン
          </Text>
          <ActivityIndicator
            animating={true}
            color="white"
            hidesWhenStopped={true}
            size="large"
            style={
              Array [
                Object {
                  "height": 0,
                  "marginHorizontal": 10,
                },
                undefined,
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`ログインフォーム snapshot-メールアドレスとパスワードが入力されている 1`] = `
<View
  style={
    Object {
      "marginTop": 140,
    }
  }
>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            Object {
              "fontSize": 20,
            },
            undefined,
          ],
        ]
      }
    >
      メールアドレス
    </Text>
  </View>
  <View
    collapsable={undefined}
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      onChangeText={undefined}
      placeholder="example@hoge.com"
      secureTextEntry={undefined}
      selectionColor="#86939e"
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          Object {
            "fontSize": 18,
          },
        ]
      }
      value="example@gmail.com"
    />
  </View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            Object {
              "fontSize": 20,
            },
            undefined,
          ],
        ]
      }
    >
      パスワード
    </Text>
  </View>
  <View
    collapsable={undefined}
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      onChangeText={undefined}
      placeholder="パスワード"
      secureTextEntry={true}
      selectionColor="#86939e"
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          Object {
            "fontSize": 18,
          },
        ]
      }
      value="PassWord"
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "color": "#FF0000",
        "fontSize": 18,
        "paddingTop": 5,
      }
    }
  >
    
  </Text>
  <View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "marginLeft": 15,
            "marginRight": 15,
          },
          Object {
            "shadowColor": "rgba(0,0,0, .4)",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          undefined,
          undefined,
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "marginBottom": 5,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "blue",
              },
              undefined,
              false,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                  undefined,
                  false,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            ログイン
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`ログインフォーム snapshot-ロード中 1`] = `
<View
  style={
    Object {
      "marginTop": 140,
    }
  }
>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            Object {
              "fontSize": 20,
            },
            undefined,
          ],
        ]
      }
    >
      メールアドレス
    </Text>
  </View>
  <View
    collapsable={undefined}
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      onChangeText={undefined}
      placeholder="example@hoge.com"
      secureTextEntry={undefined}
      selectionColor="#86939e"
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          Object {
            "fontSize": 18,
          },
        ]
      }
      value="example@gmail.com"
    />
  </View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            Object {
              "fontSize": 20,
            },
            undefined,
          ],
        ]
      }
    >
      パスワード
    </Text>
  </View>
  <View
    collapsable={undefined}
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      onChangeText={undefined}
      placeholder="パスワード"
      secureTextEntry={true}
      selectionColor="#86939e"
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          Object {
            "fontSize": 18,
          },
        ]
      }
      value="PassWord"
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "color": "#FF0000",
        "fontSize": 18,
        "paddingTop": 5,
      }
    }
  >
    
  </Text>
  <View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "marginLeft": 15,
            "marginRight": 15,
          },
          Object {
            "shadowColor": "rgba(0,0,0, .4)",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          undefined,
          undefined,
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "marginBottom": 5,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "blue",
              },
              undefined,
              false,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                  undefined,
                  false,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            ログイン
          </Text>
          <ActivityIndicator
            animating={true}
            color="white"
            hidesWhenStopped={true}
            size="large"
            style={
              Array [
                Object {
                  "height": 0,
                  "marginHorizontal": 10,
                },
                undefined,
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`ログインフォーム snapshot-普通 1`] = `
<View
  style={
    Object {
      "marginTop": 140,
    }
  }
>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            Object {
              "fontSize": 20,
            },
            undefined,
          ],
        ]
      }
    >
      メールアドレス
    </Text>
  </View>
  <View
    collapsable={undefined}
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      onChangeText={undefined}
      placeholder="example@hoge.com"
      secureTextEntry={undefined}
      selectionColor="#86939e"
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          Object {
            "fontSize": 18,
          },
        ]
      }
      value={undefined}
    />
  </View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            Object {
              "fontSize": 20,
            },
            undefined,
          ],
        ]
      }
    >
      パスワード
    </Text>
  </View>
  <View
    collapsable={undefined}
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      onChangeText={undefined}
      placeholder="パスワード"
      secureTextEntry={true}
      selectionColor="#86939e"
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          Object {
            "fontSize": 18,
          },
        ]
      }
      value={undefined}
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "color": "#FF0000",
        "fontSize": 18,
        "paddingTop": 5,
      }
    }
  />
  <View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "marginLeft": 15,
            "marginRight": 15,
          },
          Object {
            "shadowColor": "rgba(0,0,0, .4)",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          undefined,
          undefined,
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "marginBottom": 5,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "blue",
              },
              undefined,
              false,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                  undefined,
                  false,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            ログイン
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
